// @codepup/iframe-router/dist/chunk-EXYKYZAQ.mjs
// src/vite.ts
function codepupPreview() {
  return {
    name: "codepup-iframe-router",
    enforce: "post",
    transformIndexHtml() {
      return [
        {
          tag: "script",
          attrs: { type: "module" },
          children: `
            (() => {
              if (window.parent === window) return;

              const send = () => {
                window.parent.postMessage(
                  {
                    source: "codepup",
                    type: "ROUTE_CHANGE",
                    path: location.pathname + location.search + location.hash
                  },
                  "*"
                );
              };

              // ---- CHILD \u2192 PARENT ----
              send();
              window.addEventListener("popstate", send);

              ["pushState", "replaceState"].forEach((key) => {
                const original = history[key];
                history[key] = function (...args) {
                  const result = original.apply(this, args);
                  send();
                  return result;
                };
              });

              // ---- PARENT \u2192 CHILD ----
              window.addEventListener("message", (event) => {
                if (event.data?.type !== "NAVIGATE") return;

                const path = event.data.path;
                if (typeof path !== "string") return;

                // Avoid duplicate navigation
                if (path === location.pathname + location.search + location.hash) return;

                history.pushState({}, "", path);
                window.dispatchEvent(new PopStateEvent("popstate"));
              });
            })();
          `.trim()
        }
      ];
    }
  };
}

export {
  codepupPreview
};
