// @codepup/iframe-router/dist/vite.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/vite.ts
var vite_exports = {};
__export(vite_exports, {
  codepupPreview: () => codepupPreview
});
module.exports = __toCommonJS(vite_exports);
function codepupPreview() {
  return {
    name: "codepup-iframe-router",
    enforce: "post",
    transformIndexHtml() {
      return [
        {
          tag: "script",
          attrs: { type: "module" },
          children: `
            (() => {
              if (window.parent === window) return;

              const send = () => {
                window.parent.postMessage(
                  {
                    source: "codepup",
                    type: "ROUTE_CHANGE",
                    path: location.pathname + location.search + location.hash
                  },
                  "*"
                );
              };

              // ---- CHILD \u2192 PARENT ----
              send();
              window.addEventListener("popstate", send);

              ["pushState", "replaceState"].forEach((key) => {
                const original = history[key];
                history[key] = function (...args) {
                  const result = original.apply(this, args);
                  send();
                  return result;
                };
              });

              // ---- PARENT \u2192 CHILD ----
              window.addEventListener("message", (event) => {
                if (event.data?.type !== "NAVIGATE") return;

                const path = event.data.path;
                if (typeof path !== "string") return;

                // Avoid duplicate navigation
                if (path === location.pathname + location.search + location.hash) return;

                history.pushState({}, "", path);
                window.dispatchEvent(new PopStateEvent("popstate"));
              });
            })();
          `.trim()
        }
      ];
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  codepupPreview
});
